compiler/optimizer/VPHandlers.cpp

```diff
@@ -447,6 +446,25 @@ static bool tryFoldCompileTimeLoad(
                }
             return didSomething;
             }
+         else if (node->getSymbolReference() == vp->comp()->getSymRefTab()->findVftSymbolRef()
+                  && node->getFirstChild() == curNode
+                  && constraint->isNonNullObject()
+                  && constraint->getClassType()
+                  && constraint->getClassType()->asFixedClass()
+                  && constraint->getClass()
+                  && !vp->comp()->compileRelocatableCode())
```

Comment:
@dsouzai does this restriction apply to SVM based AOT code too ?

Reply:
It should be fine to do this transformation with the SVM. It looks like it's also OK in old-style AOT if the class was loaded by the bootstrap loader

Reply:
> It should be fine to do this transformation with the SVM. It looks like it's also OK in old-style AOT if the class was loaded by the bootstrap loader
Yes, that was what I had in mind as we do add relocations. But in testing I bumped into case where we did this transformation for the vft-load node under virtual call (Loading method pointer from the v-table off the J9Class pointer) and we got the dereferenced method pointer which does not point to correct location.

Reply:
Thanks a lot @jdmpapin for the review. You are correct, the failure went away when I set the CP index to -1. Changes made in https://github.com/eclipse/omr/compare/c35d09156ee864be4bac1719ab8570f4d8a88d04..8066b3dfe38dd091f069482ff4d5daf4cefce3f8 makes sure, that the we do the transformation in Old AOT only when the class is loaded by system class loader.